app/javascript/services/instrument.service.js

```diff
@@ -0,0 +1,32 @@
+const AudioContext = window.AudioContext || window.webkitAudioContext;
+
+export default class InstrumentService {
+    constructor(sample) {
+        this.sample = sample
+    }
+
+    play(rate) {
+        const context = new AudioContext();
+        const source = context.createBufferSource();
+        source.buffer = this.sample;
+        source.playbackRate.value = rate;
+        source.connect(context.destination);
+        source.start(0);
+    }
+
+    static async loadSample(url) {
+        const context = new AudioContext()
+        if (context.decodeAudioData.length !== 1) {
+            const originalDecodeAudioData = context.decodeAudioData.bind(context);
+            context.decodeAudioData = buffer =>
+                new Promise((resolve, reject) =>
+                    originalDecodeAudioData(buffer, resolve, reject)
+                );
+        }
+
+        const response = await fetch(url)
+        const buffer = await response.arrayBuffer()
+
+        return await context.decodeAudioData(buffer)
+    }
+}
```

Comment:
Should we move this into `sound-engines` — IMO this is exactly a sound engine class :D

Reply:
In fact, I think you can bring all this code inside the `SamplerEngine` class directly :) 

Because an instrument based on an audio file is called a `sampler` 😇

Reply:
@malparty so what is the purpose of `changeInstrument()` function? 🤔 
Changing notes?

Reply:
Nope: The note is provided with the `onPadPressed` function (we could later eventually add a `rate` param so that we can centralize the notes and scale).

The `changeInstrument` is IF you want to use different audio files (with audio from another instrument, like a guitar, a piano, etc...). Calling the `changeInstrument` should 
- switch to the next audio file (assuming we have an array of audio files)
- update the `this.instrument` with the name of this audio file (e.g. `Guitar`, `Piano`, ...)

Reply:
Updated at [81eb0e6](https://github.com/malparty/nimbleat/pull/16/commits/81eb0e68581f202da3f65ca436380cc260e64416)